src/cli/cli.cpp

```diff
@@ -1716,7 +1716,20 @@ void Interpreter::ProcessPskc(uint8_t aArgsLength, char *aArgs[])
     {
         otPskc pskc;
 
-        VerifyOrExit(Hex2Bin(aArgs[0], pskc.m8, sizeof(pskc)) == OT_PSKC_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
+        if (aArgsLength == 1)
+        {
+            VerifyOrExit(Hex2Bin(aArgs[0], pskc.m8, sizeof(pskc)) == OT_PSKC_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
+        }
+        else if (!strcmp(aArgs[0], "-p"))
+        {
+            otDatasetGeneratePskc(aArgs[1], otThreadGetNetworkName(mInstance), otThreadGetExtendedPanId(mInstance),
```

Comment:
In the case that the dataset includes a Thread Network Name and/or Extended PAN ID, I think it is confusing that the PSKc comes from the current state in the stack. How about extracting the value of those fields if they are already defined in the dataset. If not, fall back to using `otThreadGetNetworkName()` and `otThreadGetExtendedPanId()`?

Reply:
otThreadGetNetworkName() and otThreadGetExtendedPanId() is the same as the values in dataset because mle restores dataset. And pskc cannot be changed after the device is already attached.

Reply:
Thie docs you wrote above confused me. Are you looking to also support updating the PSKc via the `dataset` CLI command?

Reply:
dataset can be changed at runtime by FTD and MTD, however MTD doesn't support computing t PSKc from passphrase. To reduce the code size of MTD, I prefer not add this into dataset command, throught?

Reply:
The recommended way for configuring Thread Network parameters is via the Operational Dataset (see [src/cli/README_DATASET.md](https://github.com/openthread/openthread/blob/master/src/cli/README_DATASET.md#quick-start)).

If we are concerned about code size, can we simply exclude the PSKc computation from MTD? If no one calls `otDatasetGeneratePskc()`, the linker should strip the dead code as well.